File: NWorldPermissions/src/top/nololiyt/worldpermissions/commands/reload/ReloadRouter.java
package top.nololiyt.worldpermissions.commands.reload;

import top.nololiyt.worldpermissions.commands.CommandLayer;
import top.nololiyt.worldpermissions.commands.Router;

import java.util.LinkedHashMap;
import java.util.Map;

public class ReloadRouter extends Router
{
    protected final static String layerName = "reload";
    
    @Override
    public String permissionName()
    {
        return layerName;
    }
    
    @Override
    public String messageKey()
    {
        return layerName;
    }
    
    private Map<String, CommandLayer> commandLayers = new LinkedHashMap<String, CommandLayer>()
    {
        {
            put("config", new ConfigExecutor());
            put("messages", new MessagesExecutor());
            put("marks", new MarksExecutor());
            put("worlds", new WorldsExecutor());
            put("all", new AllExecutor());
        }
    };
    
    @Override
    protected Map<String, CommandLayer> nextLayers()
    {
        return commandLayers;
    }
}

File: NWorldPermissions/src/top/nololiyt/worldpermissions/commands/reload/ReloadRouter.kt
package top.nololiyt.worldpermissions.commands.reload

import top.nololiyt.worldpermissions.commands.CommandLayer
import top.nololiyt.worldpermissions.commands.Router

class ReloadRouter : Router()
{
    override fun permissionName(): String?
    {
        return layerName;
    }
    
    override fun messageKey(): String?
    {
        return layerName;
    }
    
    /**
     * Return the next layer matching the arg.
     * If no layer match it, please return 'null' and the help list will be sent.
     *
     * @param arg
     * @return
     */
    override fun nextLayer(arg: String): CommandLayer?
    {
        return when (arg)
        {
            "config"   -> ConfigExecutor()
            "messages" -> MessagesExecutor()
            "marks"    -> MarksExecutor()
            else       -> null
        };
    }
    
    companion object
    {
        private const val layerName= "reload";
    }
}